pkg/bundle/read.go

```diff
@@ -64,6 +64,27 @@ func Open(ctx context.Context, name, baseDir, file string, opts *Options) (*Bund
 	return Read(ctx, name, baseDir, in, opts)
 }
 
+// Try accessing the documented, primary fleet.yaml extension first. If that returns a "IsNotExist" error, then we
+// try the fallback extension. If we receive "IsNotExist" errors for both file extensions, then we return a "nil" file
+// and a "nil" error. If either return a non-"IsNotExist" error, then we return the error immediately.
+func setupIOReader(baseDir string) (*os.File, error) {
+	if file, err := os.Open(bundleyaml.GetFleetYamlPath(baseDir, false)); err != nil && !os.IsNotExist(err) {
+		return nil, err
+	} else if err == nil {
+		defer file.Close()
```

Comment:
You will close file once this method is returned, so the reader can't be read anymore after it has been returned

Reply:
Good catch